pkg/nodediscovery/nodediscovery.go

```diff
@@ -207,6 +207,15 @@ func (n *NodeDiscovery) StartDiscovery(nodeName string) {
 		})
 	}
 
+	// Add the tunnel endpoint address here
+	log.Debug("Adding local node tunnel endpoint address")
+	if node.GetTunnelEndpointIPv4() != nil {
```

Comment:
Between K8sExternalIPv4, IPv4, IPv4Router, TunnelIPv4 I'm getting confused by which IP addresses will be used in which circumstance. Do we really need to keep track of as many as four different IP addresses here?

Reply:
/cc @AnishShah any thoughts here? Do you see semantic overlap with the existing address fields?

Reply:
My gut feeling on first reading this is that the option you're trying to specify here should be able to configure the `NodeInternalIP` and let those codepaths handle the traffic rather than adding a new dedicated tunnel option which is sometimes specified, sometimes not.

Reply:
Thanks for questions here. I think each address type here is an needed one. Maybe we can extend the use of TunnelIPv4 here to separate the data path and control path in the future. But it actually provide us something now.
Other comments:
https://github.com/cilium/cilium/pull/15072#issuecomment-796542640